config/initializers/hyrax-doi.rb

```diff
@@ -168,3 +168,18 @@ def assign_for(object:, attribute: :identifier)
     object
   end
 end
+
+# Forces DataCiteRegistrar to initializew a new client everytime. 
+# A fix for #245 https://github.com/scientist-softserv/britishlibrary/issues/245
+# Prevents leaching of datacite config between tenants when minting DOIs by ensuring
+# that the datacite credentials always come from the account that kicked the job off
+Hyrax::DOI::DataCiteRegistrar.class_eval do
+
+  def client 
+    current_tenant = Apartment::Tenant.current
+    @datacite_endpoint ||= Account.find_by(tenant: current_tenant).data_cite_endpoint
```

Comment:
I'm wondering if the prefix causes problems as well?  https://github.com/samvera-labs/hyrax-doi/blob/main/app/services/hyrax/doi/datacite_registrar.rb#L10-L12

Also, I think we may want to consider memoize the entire client:

```ruby
def client
  @client ||= begin
    current_tenant = Apartment::Tenant.current
    datacite_endpoint = Account.find_by(tenant: current_tenant).data_cite_endpoint
    Hyrax::DOI::DataCiteClient.new(username: datacite_endpoint.username, password: datacite_endpoint.password, prefix: datacite_endpoint.prefix, mode: datacite_endpoint.mode)
  end
end
```

Also a `class_attribute` can be set in the instance (recommended during an initialize).  So we could do `self.password = <client's data cite end point password>`.

Reply:
All excellent points! The prefix in the initialize wasn't causing observable issues for me, but could well do in some contexts. Let me see if I can knock this in to a better shape...